### ISSUE TITLE
Feedback & Suggestions in v1.1.0-alpha

### ISSUE THREAD

**Issue by SamejSpenser** [bogosorter/elementary#1]
Hey bogosorter,

I mentioned on Reddit that I'd bring some feedback, so here it is, as you asked:

- I was really pleasantly surprised to see your thank-you note in the `v1.1.0-alpha` release. I didn't expect that and was super happy to see my name mentioned. It's a pleasure to help out with your project by giving my humble opinions as a markdown editor user.

- I really liked the preview mode; it's well-made and fits nicely with the program itself.

- If I can make more suggestions, maybe a little pop-over with the button name/function could show up when hovering over them (in both edit and preview modes). At first glance, I didn't realize the binocular icon was for preview and only got it after reading your notes. I guess I'm used to the eye or glasses icon in other editors, but a pop-over explaining each button would make it way easier to understand.

- While working on a text like I usually do, I missed having Elementary "jump" between footnotes and back to the text when clicking the note. The superscript number and return arrow look like links, but nothing happens when I click them.

- I’m not a fan of seeing the YAML front matter in preview mode (even if it’s like GitHub does it). I’d prefer if the YAML part was hidden in preview mode, showing only from the note’s title onward.

- Using the find-and-replace tool (`Ctrl + H`), I missed seeing how many replacements were made when hitting "Replace All."

- I’d love it if the cursor stayed in the same spot when switching between edit and preview modes. Switching sends it back to the top, and then I have to scroll all the way down to where I left off.

- It’d also be great if the line with the cursor stayed highlighted. It’d help us find our place in the text faster (especially for folks like me who are nearsighted 😅).

- In preview mode, I’d like to choose if the text is aligned justified, left, or right (either in settings or with inline tags or embedded CSS I use in my notes).

- Large images (wider than the set width) don’t stay within the width and spill over the side of the text area. I use CSS to keep image widths variable, ranging from 15% to 100% of the text area width, and as shown in the screenshot, the image spills out.

These are the details I noticed while using Elementary for one of my study notes, like I would with Pulsar, Sublime Text, or Micro Editor.

Some features are really awesome, like the color scheme harmony, the small highlights when hovering over buttons, the smooth cursor movement, and the quick response of shortcuts and button presses. It’s clear a lot of work and dedication went into making Elementary a fantastic program.

If the improvements keep up like they did from `v1.0.0` to `v1.1.0-alpha`, it’s easily going to become my go-to editor! 😉

<p>&nbsp;</p>

### Some screenshots

![Screenshot 01](https://telegra.ph/file/fa6a565f6419d44168563.jpg)  
<small>Screenshot 01: YAML front matter in preview mode</small>

<p>&nbsp;</p>

![Screenshot 02](https://telegra.ph/file/0524ce0d23773d437450a.jpg)  
<small>Screenshot 02: YAML front matter in editor mode</small>

<p>&nbsp;</p>

![Screenshot 03](https://telegra.ph/file/269d1214744204dd237d2.jpg)  
<small>Screenshot 03: Large image in preview mode</small>

<p>&nbsp;</p>

![Screenshot 04](https://telegra.ph/file/b4ac315f70a2783701dd1.jpg)  
<small>Screenshot 04: Large image in editor mode</small>

<p>&nbsp;</p>

**Comment by bogosorter** [comment]
Hi! It's a pleasure to talk to you again :)

Here are my thoughts on your suggestions:

- **Preview icon:** I've changed the binoculars icon into an eye. Although I prefer the binoculars, consistency with the norm and simplicity of use are more important than my idiosyncrasies...
- **Tooltips:** I've added tooltips to all the buttons on the sidebar.
- **Footnotes:** That was indeed a bug in the preview mode. Fixed :)
- **YAML front matter in preview mode:** Gone :)
- **Find-and-replace:** To be honest, I don't fully understand what you mean. When you search for a word in the find-and-replace tool, there is a counter for the number of occurrences to the right (see screenshot below). Replacing all will replace all of those occurrences.
- **Loosing track of text while switching between edit and preview mode:** I'll separate this in two separate tasks:

    1) **Maintaining the selection/cursor and scroll:** The selection (i.e., the cursor if the selection is empty) is now persistent across preview toggles. When Elemtary switches back into editing mode the previous selection is focused and scrolled to the middle of the editor.
    2) **Coordinating editor and preview modes:** This is a more challenging task. Given the current structure of the app, there isn't a direct mapping between a given line number and a value for scroll in preview mode (and vice versa). I'll think more about this and try to find a solution.

- **Highlighting the line with the cursor:** I just added a setting that does that. It's set to false by default.
- **Choosing text alignment in preview mode:** There another setting for that now :)
- **Large images spilling over:** That's fixed too. I've set the maximum width to 100% of the preview area, which can be overriden with inline CSS.
- **Custom CSS and local files:** If I understood correctly, you were trying to load a style sheet, which wasn't working. In fact, custom CSS is working as long as you embed it directly in the document or load it from an external source. What's _not_ working is loading CSS from a local file. The same applies to local images. This is due to security measures in Electron. I'll add this to the roadmap.

Just a final note: the thank-you note was well deserved. I try not to come across as too hyperbolic since that can diminish the value of future compliments, but I'm genuinely impressed by you effort and the hours you've put into this. Thank you!

I'll work on the issues that aren't resolved yet and keep you updated :)

![Screenshot from 2024-08-17 19-35-29](https://github.com/user-attachments/assets/c5735373-adf1-46ce-ad45-9360e1d67cb3)

**Comment by SamejSpenser** [comment]
Hey again. 😉

- **Find and replace:** I just realized it was my oversight; I'm used to getting feedback on the number of changes after they're done (that's the standard in Sublime Text and Micro Editor), and I didn't notice the info was there before replacing; sorry about that.

- **About custom CSS:** I usually use my custom CSS in my markdown notes by inserting the lines below right after the YAML front matter:

    ```md
    <!-- LINK EXTERNO PARA O CSS -->
    
    <link rel="stylesheet" type="text/css" href="/home/samej/Sync/Obsidian/GERAIS/Gists-Privadas/CSS-geral/stylesheet-geral.css" />
    
    <link rel="stylesheet" type="text/css" href="https://gitlab.com/samej/samej/-/blob/fa32cbc9864c98d87395841d4df505e7ed5929ef/stylesheet-geral.css" />
    
    <link rel="stylesheet" type="text/css" href="https://www.dropbox.com/s/7acsnogog4njf2o/stylesheet-geral.css" />
    
    <!-- LINK EXTERNO PARA O CSS NO SMARTPHONE -->
    
    <link rel="stylesheet" type="text/css" href="/storage/emulated/0/Obsidian/.obsidian/snippets/stylesheet-geral.css" />
    
    ```
    
    This way, I can pretty much ensure uniformity in my notes, aligning images, embeds, decorations, etc.

    I’m not sure if it’s worth adding the option to include custom CSS in Elementary. I think its simplicity and focus are great as they are; but if it could respect the CSS like in the example above, I’d be really happy! 😉

- I'm really glad you appreciate my feedback; it's a pleasure to collaborate on a project like yours.

I’m eagerly waiting for new releases to test the upcoming changes, and thank you in advance for working on the suggestions I make. But take all the time you need to make the changes.

**Comment by bogosorter** [comment]
Hi! I've been away for a while, but now I'm back to work :)

I just uploaded a new [release](https://github.com/bogosorter/elementary/releases/tag/v1.1.0-beta.0) with the bug fixes and improvements I mentioned earlier. It also features support for local images and CSS.

See you soon!

**Comment by SamejSpenser** [comment]
Hey again, @bogosorter.

I tested `Elementary v1.1.0-beta.0` this morning and it was awesome! I didn’t run into any problems.

The local images and CSS worked perfectly. The shortcuts did what they were supposed to, the overall smoothness of the program is excellent, and the icons and commands in the sidebar are clearly showing what they do.

Anyway, for a program that aims to be a simple markdown editor, it’s doing great and totally nailed its goal.

The only suggestion I have right now is to revise the text in the Elementary description (about) and the "Update notice."

If it’s not too much to ask, and I’m not pushing my luck, maybe add a few more keyboard shortcuts, like `Ctrl + D` to duplicate a line, `Ctrl + U` and `Ctrl + Shift + U` for UPPER CASE and lower case, respectively.

Those shortcuts would really help me out with typing and editing my markdown notes. 😉

That’s all I have to report for now.

Best,

**Comment by bogosorter** [comment]
Hi again, @SamejSpenser!

As always, it's a pleasure to read your feedback.

I just published the final `v1.1.0` release. Thank you for helping me go through this :)

I've been working on PDF exporting and expect a new alpha release in the next few days. I think that's not something you're too interested in, but I'll make sure to include those shortcuts ;)

Cheers

**Comment by bogosorter** [comment]
Hi! I promised to deliver a new release in a few days but other things got in the way. I should've known better... Anyway, I won't be as fast as I'd like to. I'll get to it eventually, though :)

Cheers!
